.circleci/config.yml

```diff
@@ -1,13 +1,59 @@
-# Use the latest 2.1 version of CircleCI pipeline process engine. See: https://circleci.com/docs/2.0/configuration-reference
+---
+
 version: 2.1
-# Use a package of configuration called an orb.
-orbs:
-  # Declare a dependency on the welcome-orb
-  welcome: circleci/welcome-orb@0.4.1
-# Orchestrate or schedule a set of jobs
+
+jobs:
+  ci:
+    # ref: https://circleci.com/docs/2.0/executor-types/?section=executors-and-images#using-docker
+    docker:
+      # Use the base circle CI image
+      - image: "cimg/base:stable"
+        auth:
+          username: DOCKERHUB_LOGIN
+          password: DOCKERHUB_PASSWORD
+    steps:
+      # TODO set up source and pip caching
+      # checkout the git repo
+      - checkout
+
+      - run:
+          name: "Run tests and code coverage"
+          command: |
+            pip install -U pip wheel
+            pip install poetry
+            poetry install
+            poetry run python groclient/lib.py -v
+            mkdir -p shippable/testresults shippable/codecoverage &&
+            poetry run pytest
+            --junitxml=shippable/testresults/nosetests.xml
+            --cov=groclient
+            --cov-report=xml:shippable/codecoverage/coverage.xml
+            pip install -r api/client/samples/analogous_years/requirements.txt &&
+            poetry run pytest api/client/samples/analogous_years/
+
+      - run:
+          name: "Run post build steps"
+          when: on_success
+          command: |
+            poetry install -E docs &&
+            poetry run sphinx-build -W --keep-going docs docs/_build/html
+            poetry run sphinx-build -b linkcheck docs docs/_build/linkcheck
+            git config --global user.email "api-documentation@gro-intelligence.com" &&
+            git config --global user.name "Gro Intelligence" &&
+            git remote set-url origin git@github.com:<< pipeline.project.git_url >> &&
+            poetry run bin/sphinx_push_ghpages.sh
+            pip install poetry-dynamic-versioning &&
+            poetry build
+            if [ "$BRANCH" == "development" -a "$IS_PULL_REQUEST" == "false" ]; then
+              poetry config repositories.testpypi https://test.pypi.org/legacy/ &&
+              poetry publish -u __token__ -p $TESTPYPI_TOKEN -r testpypi
+            fi
+            if [ "$IS_RELEASE" == "true" ]; then
+              poetry publish -u __token__ -p $PYPI_TOKEN
```

Comment:
I think we should probably avoid doing this (for now) since it'll conflict with Shippable?

Publishing to testpypi seems fine, though.

Reply:
Also, how are the TESTPYPI_TOKEN and PYPI_TOKEN env vars getting set in CircleCI?

Reply:
This was just a debugging draft of the PR, should have done a better job of putting that in the description. I haven't yet set up the environment variables, but I'm planning to use the following measures for open source projects' variables: https://circleci.com/docs/2.0/oss/

Reply:
Removed the line to publish to pypi, leaving testpypi.